### PR TITLE
Integration test against final artefact through wrangler.

### PR DESCRIPTION
**Pull request by tomlarkworthy**
Added an extra CI step to conform example README instruction work by locally deploying the script in wrangler.

I confirmed failing the test also fails the job here https://github.com/tomlarkworthy/cloudworker-proxy/actions/runs/6135617088/job/16649542963?pr=3

fixes https://github.com/markusahlstrand/cloudworker-proxy/issues/129

Now that we have end to end testing against a final artefact we can consider switching to bun build and moving all source code to typescript, and we can test the final JS bundle works after building by using this integration test.

While I was here I switched to `bun install` over `npm install` as its much faster, removing the package-lock in preference to the binary bun.lockb. Bun has reached v1! https://bun.sh/blog/bun-v1.0

### CODE REVIEW

**Comment by tomlarkworthy** [comment]
base branch https://github.com/markusahlstrand/cloudworker-proxy/pull/125

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  pull_request:
+    branches: [master]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        bun-version: [14.x]
```
Comment (markusahlstrand) [inline .github/workflows/node.js.yml:16]: Should this be 1.0.0 as well?

**Comment by markusahlstrand** [comment]
This is really cool! I've been wanting to do this but never got around to it. Looking forward to see how it works in practice.

File: src/handlers/s3.js [hunk 2]
```diff
@@ -1,15 +1,50 @@
 const { AwsClient } = require('aws4fetch');
 const utils = require('../utils');
 
-function s3HandlerFactory({ accessKeyId, secretAccessKey, bucket, region }) {
+function getEndpoint(endpoint, options) {
```
Comment (markusahlstrand) [inline src/handlers/s3.js:4]: I've been having issues with code missing the last couple of weeks when doing PR's between branches. This code was merged in the S3-pr right?

Comment (markusahlstrand) [reply src/handlers/s3.js]: Update the branch now and it looks fine!

**Pull request merged**